### PR TITLE
Initial work

### PR DESCRIPTION
**Pull request by hanpeter**
Thin wrapper for EC2, just like [python-krux-boto-sqs](https://github.com/krux/python-krux-boto-sqs).

98% of the code in `krux_ec2/ec2.py` and `krux_ec2/filter.py` is copy and paste from [python-krux-manage-instance](https://github.com/krux/python-krux-manage-instance/blob/master/krux_ec2/ec2.py), and thus reviewed before.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,38 @@
 # krux_ec2
+
+`krux_ec2` is a library that provides wrapper functions for common EC2 usage. It uses `krux_boto` to connect to AWS EC2.
+
+## Warning
+
+In the current version, `krux_ec2.ec2.EC2` is only compatible with `krux_boto.boto.Boto` object. Passing other objects, such as `krux_boto.boto.Boto3`, will cause an exception.
```
Comment (dcrampton) [inline README.md]: Is there a plan to allow for boto3 in the future?

File: README.md [hunk 2]
```diff
@@ -1 +1,38 @@
...
+
+## Application quick start
+
+The most common use case is to build a CLI script using `krux_boto.cli.Application`.
+Here's how to do that:
```
Comment (dcrampton) [inline README.md]: Are these comments specific to this pull request, or to the files that it's built upon?

File: krux_ec2/ec2.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
+# -*- coding: utf-8 -*-
+#
+# © 2015 Krux Digital, Inc.
+#
+
+# TODO: This is currently inside krux_manage_instance library.
+# However, consider breaking this into a separate library or add it to krux_boto library.
```
Comment (dcrampton) [inline krux_ec2/ec2.py]: Doing this TODO right now?

File: krux_ec2/ec2.py [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+            raise TypeError('krux_ec2.ec2.EC2 only supports krux_boto.boto.Boto')
+
+        self.boto = boto
+
+        # Set up default cache
+        self._conn = None
+
+    def _get_connection(self):
+        """
+        Returns a connection to the designated relsgion (self.boto.cli_region).
```
Comment (dcrampton) [inline krux_ec2/ec2.py]: tyop.  "relsgion" should be "region"

File: krux_ec2/ec2.py [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+        Starts an instance in the given AMI.
+
+        The ami_id parameter is the ID of the AMI image where the new instance will be created.
+        The cloud_config parameter is passed as the user data.
+        The instance_type, sec_group, and zone are passed as respective parameters to AWS.
+
+        4 block devices are created as ephemeral and passed.
+        """
+        # OK, so, on larger instances, extra devices only show up if you tell them to associate with a block device.
+        # These EBS AMIs don't set this up, so we have to. sdb will always be ephemeral0,
```
Comment (dcrampton) [inline krux_ec2/ec2.py]: tyop, extra zero at the end of ephemeral

**Comment by dcrampton** [comment]
Looks good, assuming tested.

File: README.md (hunk 1 above)
Comment (hanpeter) [inline README.md]: Yep. Next major change to this library should be that.

File: krux_ec2/ec2.py (hunk 5 above)
Comment (hanpeter) [inline krux_ec2/ec2.py]: That actually means "ephemeral0" at [Line 202](https://github.com/krux/python-krux-boto-ec2/pull/1/files#diff-5ca09400edfb66ee9ab426507588761fR202).

**Pull request merged**